### PR TITLE
1448897: Enforce default content access mode list

### PR DESCRIPTION
**Pull request by wottop**
If no list exists, then it is set to the default list and
 the mode is also set to the default value ['entitlement']
Ensure that the org's content access mode is defined when the
 org is created or updated
Update the mode list and mode columns to not null and default.

### CODE REVIEW

**Comment by wottop** [comment]
buildbot: retest this please

**Comment by wottop** [comment]
buildbot: retest this please

File: server/src/main/java/org/candlepin/controller/OwnerManager.java [hunk 1]
```diff
@@ -57,26 +57,54 @@
...
+        this.ownerCurator = ownerCurator;
+        this.contentAccessCertService = contentAccessCertService;
+        this.contentAccessCertCurator = contentAccessCertCurator;
+        this.ownerEnvContentAccessCurator = ownerEnvContentAccessCurator;
+        this.uberCertificateCurator = uberCertificateCurator;
+        this.ownerServiceAdapter = ownerServiceAdapter;
+    }
     @Transactional
-    public void cleanupAndDelete(Owner owner, boolean revokeCerts) {
+    public void cleanupAndDelete(Owner owner, PoolManager poolManager, boolean revokeCerts) {
```
Comment (Ceiu) [inline server/src/main/java/org/candlepin/controller/OwnerManager.java]: Why do we not inject the pool manager anymore?

File: server/src/main/java/org/candlepin/controller/OwnerManager.java [hunk 2]
```diff
@@ -198,19 +226,18 @@ public void refreshContentAccessMode(OwnerServiceAdapter adapter, Owner owner) {
...
         // This shouldn't happen, but in the event our upstream source is having issues, let's
         // not put ourselves in a bad state as well.
-        if (upstreamList != null && !upstreamList.isEmpty()) {
+        if (!StringUtils.isBlank(upstreamList)) {
             // Not empty list. Verify that the upstream mode is present.
             String[] list = upstreamList.split(",");
-
-            if (!StringUtils.isEmpty(upstreamMode) && !ArrayUtils.contains(list, upstreamMode)) {
-                throw new IllegalStateException("Upstream access mode is not present in access mode list");
+            if (upstreamMode == null || !ArrayUtils.contains(list, upstreamMode)) {
```
Comment (Ceiu) [inline server/src/main/java/org/candlepin/controller/OwnerManager.java]: Are we no longer treating an empty string the same as a null value? What happens if upstream gives us an empty string here?

File: server/src/main/java/org/candlepin/resource/ConsumerResource.java [hunk 3]
```diff
@@ -1518,9 +1515,9 @@ public Response getContentAccessBody(
             throw new BadRequestException(i18n.tr("Content access body " +
                "can not be requested for a share consumer"));
         }
-        if (consumer.getOwner().contentAccessMode()
-            .equals(ContentAccessCertServiceAdapter.DEFAULT_CONTENT_ACCESS_MODE)) {
-            throw new BadRequestException(i18n.tr("No content access mode assigned"));
+        if (!consumer.getOwner().getContentAccessMode()
```
Comment (Ceiu) [inline server/src/main/java/org/candlepin/resource/ConsumerResource.java]: The output of getContentAccessMode can be null now.

File: server/src/main/java/org/candlepin/resource/OwnerResource.java [hunk 4]
```diff
@@ -948,17 +963,17 @@ public Owner updateOwner(@PathParam("owner_key") @Verify(Owner.class) String key
             toUpdate.setAutobindDisabled(owner.getAutobindDisabled());
         }
 
-        if (!ContentAccessCertServiceAdapter.DEFAULT_CONTENT_ACCESS_MODE.equals(owner.contentAccessMode()) &&
-            config.getBoolean(ConfigProperties.STANDALONE)) {
-
+        if (config.getBoolean(ConfigProperties.STANDALONE) &&
+            !toUpdate.getContentAccessMode().equals(owner.getContentAccessMode()) &&
```
Comment (Ceiu) [inline server/src/main/java/org/candlepin/resource/OwnerResource.java]: Need a null check here.

File: server/src/main/java/org/candlepin/controller/OwnerManager.java (hunk 1 above)
Comment (wottop) [inline server/src/main/java/org/candlepin/controller/OwnerManager.java]: Did it for the unit test. Needed to mock.

File: server/src/main/java/org/candlepin/controller/OwnerManager.java (hunk 2 above)
Comment (wottop) [inline server/src/main/java/org/candlepin/controller/OwnerManager.java]: We are no longer allowing the owner fields for contentAccessMode and contentAccessModeList to be null or blank

File: server/src/main/java/org/candlepin/controller/OwnerManager.java (hunk 1 above)
Comment (Ceiu) [inline server/src/main/java/org/candlepin/controller/OwnerManager.java]: Is there another way to do this that isn't completely painful? I'm not a huge fan of rearranging production code to appease testing frameworks.

**Comment by Ceiu** [comment]
I've updated a comment and replied above, but in general I believe I understand the intent here. However, I would still prefer that we take some steps to ensure that we can't crash out in the event something from our upstream or our backend data sources end up null.

**Comment by Ceiu** [comment]
ACK

**Pull request merged**